engine/apps/api/permissions.py

```diff
@@ -279,6 +281,48 @@ def has_object_permission(self, request: AuthenticatedRequest, view: ViewSetOrAP
         return True
 
 
+def user_is_authorized_basic_role(user: "User", required_permission: LegacyAccessControlRole) -> bool:
+    """
+    This function checks user basic role
+
+    user - The user to check permissions for
+    required_permission - A basic role that a user must have to be considered authorized
+    """
+    return user.role <= required_permission
```

Comment:
I would suggest trying to reuse the less-than-or-equal-to auth logic that is defined above in [`user_is_authorized`](https://github.com/grafana/oncall/pull/3173/files#diff-83058af8678d629be3d878a5947537c15c01abf6ef2eb7ba31d08b8febda0853R120-R132) (or somehow refactoring that function such that we don't need two separate ones like this)